contracts/WETH10.sol

```diff
@@ -9,85 +9,64 @@ interface ERC677Receiver {
 }
 
 interface FlashMinterLike {
-    function executeOnFlashMint(uint, bytes calldata) external;
+    function executeOnFlashMint(bytes calldata) external;
 }
 
 /// @dev WETH10 is an Ether ERC20 wrapper. You can `deposit` Ether and obtain Wrapped Ether which can then be operated as an ERC20 token. You can
 /// `withdraw` Ether from WETH10, which will burn Wrapped Ether in your wallet. The amount of Wrapped Ether in any wallet is always identical to the
 /// balance of Ether deposited minus the Ether withdrawn with that specific wallet.
 contract WETH10 {
-    string public constant name = "Wrapped Ether";
-    string public constant symbol = "WETH";
+    string public constant name = "Wrapped Ether v10";
+    string public constant symbol = "WETH10";
     uint8  public constant decimals = 18;
-    bytes32 public immutable DOMAIN_SEPARATOR;
-    bytes32 public immutable PERMIT_TYPEHASH = keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
 
-    /// @dev Emitted when allowance of `spender` for `owner` account WETH10 token is set by call to {approve}. `value` is new allowance.
+    // keccak256("Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)");
+    bytes32 public immutable PERMIT_TYPEHASH = 0x6e71edae12b1b97f4d1f60370fef10105fa2faae0126114a169c64845d6126c9;
```

Comment:
i think the reason it didn't save any gas is because you used immutable, rather than constant

Reply:
I must have skipped a step somewhere, but there seems to be no difference in runtime cost, and an increase in deployment cost, byt using `constant`.

```
·--------------------------------------------|---------------------------|---------------|----------------------------·
|            Solc version: 0.7.0             ·  Optimizer enabled: true  ·  Runs: 20000  ·  Block limit: 9500000 gas  │
·············································|···························|···············|·····························
|  Methods                                                                                                            │
····················|························|·············|·············|···············|··············|··············
|  Contract         ·  Method                ·  Min        ·  Max        ·  Avg          ·  # calls     ·  eur (avg)  │
····················|························|·············|·············|···············|··············|··············
|  WETH10           ·  permit                ·          -  ·          -  ·        71777  ·           1  ·          -  │
····················|························|·············|·············|···············|··············|··············
|  Deployments                               ·                                           ·  % of limit  ·             │
·············································|·············|·············|···············|··············|··············
|  WETH10                                    ·          -  ·          -  ·      1841986  ·      19.4 %  ·          -  │
·--------------------------------------------|-------------|-------------|---------------|--------------|-------------·
```

Reply:
sorry, to be clear, i think this result makes sense (no gas saved) because the original code uses immutable instead of constant. if the previous code has used constant, it might have saved some gas. really unclear to me why immutable is cheaper than constant for deployment.